src/test/java/org/magicdgs/readtools/cmd/argumentcollections/FixBarcodeAbstractArgumentCollectionTest.java

```diff
@@ -0,0 +1,251 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.magicdgs.readtools.cmd.argumentcollections;
+
+import org.magicdgs.readtools.utils.read.transformer.barcodes.FixRawBarcodeTagsReadTransformer;
+import org.magicdgs.readtools.utils.read.transformer.barcodes.FixReadNameBarcodesReadTransformer;
+import org.magicdgs.readtools.utils.tests.BaseTest;
+
+import org.broadinstitute.barclay.argparser.ArgumentCollection;
+import org.broadinstitute.barclay.argparser.CommandLineException;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgram;
+import org.broadinstitute.hellbender.cmdline.TestProgramGroup;
+import org.broadinstitute.hellbender.transformers.ReadTransformer;
+import org.broadinstitute.hellbender.utils.test.ArgumentsBuilder;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+/**
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public class FixBarcodeAbstractArgumentCollectionTest extends BaseTest {
+
+    @CommandLineProgramProperties(
+            oneLineSummary = "Fix barcodes CLP test",
+            summary = "Fix barcodes CLP test",
+            programGroup = TestProgramGroup.class)
+    private static class FixBarcodesCLP extends CommandLineProgram {
+        @ArgumentCollection
+        public FixBarcodeAbstractArgumentCollection args;
+
+        public FixBarcodesCLP(final boolean fixQuals) {
+            this.args = FixBarcodeAbstractArgumentCollection.getArgumentCollection(fixQuals);
+        }
+
+        @Override
+        public String[] customCommandLineValidation() {
+            args.validateArguments();
+            return null;
+        }
+
+        @Override
+        protected Object doWork() {
+            return args.getFixBarcodeReadTransformer().getClass();
+        }
+    }
+
+    @DataProvider(name = "badArgs")
+    public Object[][] getBadArguments() {
```

Comment:
Use an iterator based provider to avoid this issue, and add with a for loop the same arguments for with/without qualities if they are the same.